src/components/VModal/VModal.vue

```diff
@@ -0,0 +1,159 @@
+<template>
+  <div>
+    <div ref="triggerContainerRef" @click="onTriggerClick">
+      <!--
+        @slot The trigger. Should be a button 99% of the time. If you need custom event handling on the trigger button, ensure bubbling is not prevented or else the dialog will not open.
+          @binding {object} a11yProps Props to v-bind to the trigger element to ensure accessibility
+          @binding {boolean} visible Whether the dialog is currently visible (open)
+      -->
+      <slot
+        name="trigger"
+        :a11y-props="triggerA11yProps"
+        :visible="visibleRef"
+      />
+    </div>
+    <VModalContent
+      :visible="visibleRef"
+      :trigger-element="triggerRef"
+      :hide-on-esc="hideOnEsc"
+      :hide-on-click-outside="hideOnClickOutside"
+      :auto-focus-on-show="autoFocusOnShow"
+      :auto-focus-on-hide="autoFocusOnHide"
+      :hide="close"
+      :aria-label="label"
+      :aria-labelledby="labelledBy"
+    >
+      <slot name="default" />
+    </VModalContent>
+  </div>
+</template>
+
+<script>
+import {
+  defineComponent,
+  ref,
+  watch,
+  reactive,
+  computed,
+} from '@nuxtjs/composition-api'
+import VModalContent from '~/components/VModal/VModalContent.vue'
+
+export default defineComponent({
+  name: 'VModal',
+  components: { VModalContent },
+  /**
+   * NB: Most of these technically default to `undefined` so that the underlying `VPopoverContent`
+   * default for each of them can take over.
+   */
```

Comment:
I was going to ask this while looking at the props, glad to see it answered right here 😆